packages/app/src/providers/h5grove/models.ts

```diff
@@ -17,39 +17,113 @@ export type H5GroveEntity =
 
 export interface H5GroveBaseEntity {
   name: string;
-  type: string;
+  kind: string;
 }
 
 export interface H5GroveGroup extends H5GroveBaseEntity {
-  type: EntityKind.Group;
+  kind: EntityKind.Group;
   children?: H5GroveEntity[];
   attributes: H5GroveAttribute[];
 }
 
 export interface H5GroveDataset extends H5GroveBaseEntity {
-  type: EntityKind.Dataset;
+  kind: EntityKind.Dataset;
   shape: number[];
-  dtype: H5GroveDtype;
+  type: H5GroveType;
   chunks: number[] | null;
   filters: Filter[] | null;
   attributes: H5GroveAttribute[];
 }
 
 export interface H5GroveSoftLink extends H5GroveBaseEntity {
-  type: 'soft_link';
+  kind: 'soft_link';
   target_path: string;
 }
 
 export interface H5GroveExternalLink extends H5GroveBaseEntity {
-  type: 'external_link';
+  kind: 'external_link';
   target_file: string;
   target_path: string;
 }
 
 export interface H5GroveAttribute {
   name: string;
   shape: number[];
-  dtype: H5GroveDtype;
+  type: H5GroveType;
 }
 
-export type H5GroveDtype = string | { [k: string]: H5GroveDtype };
+export type H5GroveType =
+  | H5GroveIntegerType
+  | H5GroveFloatType
+  | H5GroveTimeType
+  | H5GroveStringType
+  | H5GroveBitfieldType
+  | H5GroveOpaqueType
+  | H5GroveCompoundType
+  | H5GroveReferenceType
+  | H5GroveEnumType
+  | H5GroveVlenType
+  | H5GroveArrayType;
+
+export interface H5GroveBaseType {
+  class: number;
+  dtype?: string;
```

Comment:
Now that all the info is contained as fields in the object `type`, it is worth keeping the `dtype` field ?

Reply:
You mean in our front-end types? Yeah probably not!

Reply:
I kept it in the back-end because I thought that some front-ends (that don't care as much about supporting all the types) might find it more convenient to use.